drivers/scheduler/k8s/k8s.go

```diff
@@ -3019,6 +2993,63 @@ func (k *K8s) addLabelsToPVC(pvc *v1.PersistentVolumeClaim, labels map[string]st
 	}
 }
 
+// CreateAutopilotRule creates the AutopilotRule object
+func (k *K8s) CreateAutopilotRule(apRule apapi.AutopilotRule) (*apapi.AutopilotRule, error) {
+	t := func() (interface{}, bool, error) {
+		apRule.Labels = defaultTorpedoLabel
+		aRule, err := k8sAutopilot.CreateAutopilotRule(&apRule)
+		if errors.IsAlreadyExists(err) {
+			if rule, err := k8sAutopilot.GetAutopilotRule(apRule.Name); err == nil {
+				logrus.Infof("Using existing AutopilotRule: %v", rule.Name)
+				return aRule, false, nil
+			}
+		}
+		if err != nil {
+			return nil, true, fmt.Errorf("Failed to create autopilot rule: %v. Err: %v", apRule.Name, err)
+		}
+
+		return aRule, false, nil
+	}
+
+	apRuleObj, err := task.DoRetryWithTimeout(t, k8sObjectCreateTimeout, DefaultRetryInterval)
+	if err != nil {
+		return nil, err
+	}
+	logrus.Infof("Created autopilot rule: %+v", apRuleObj)
+
+	return apRuleObj.(*apapi.AutopilotRule), nil
+}
+
+// GetAutopilotRule gets the AutopilotRule for the provided name
+func (k *K8s) GetAutopilotRule(name string) (*apapi.AutopilotRule, error) {
+	return k8sAutopilot.GetAutopilotRule(name)
```

Comment:
as far as i see this function is called from inside the driver only. The separate function and changing the api is not needed at this point, just call k8sAutopilot.GetAutopilotRule(name) from the UpdateRule function directly